File: app/src/main/java/agi/erecreditsmanager/MultiMajorDialogLayout.java
package agi.erecreditsmanager;

import android.content.Context;
import android.view.LayoutInflater;
import android.widget.LinearLayout;
import android.widget.RadioButton;
import android.widget.RadioGroup;

public class MultiMajorDialogLayout extends LinearLayout {

    Context context;
    RadioGroup multiMajorRadioGroup;

    public MultiMajorDialogLayout(Context context) {
        super(context);
        this.context = context;

        init();
    }

    public void init() {
        LayoutInflater inflater = (LayoutInflater) context.getSystemService(Context.LAYOUT_INFLATER_SERVICE);
        inflater.inflate(R.layout.multimajordialog, this, true);

        multiMajorRadioGroup = (RadioGroup) findViewById(R.id.multiMajorRadioGroup);
    }

    public String checkSelectedMajor() {
        int id = multiMajorRadioGroup.getCheckedRadioButtonId();
        RadioButton selectedMajorButton = (RadioButton) findViewById(id);
        return selectedMajorButton.getText().toString();
    }
}

File: app/src/main/java/agi/erecreditsmanager/MultiMajorDialogLayout.kt
package agi.erecreditsmanager

import android.content.Context
import android.view.LayoutInflater
import android.widget.Button
import android.widget.LinearLayout
import kotlinx.android.synthetic.main.multimajordialog.view.*

class MultiMajorDialogLayout(context : Context) : LinearLayout(context) {
    init {
        LayoutInflater.from(context).run {
            inflate(R.layout.multimajordialog, this@MultiMajorDialogLayout, true)
            multiMajorRadioGroup
        }
    }

    fun checkSelectedMajor() : String {
        val id = multiMajorRadioGroup.checkedRadioButtonId
        val selectedMajorButton = findViewById<Button>(id)
        return selectedMajorButton.text.toString()
    }
}